lang/mrustc/Portfile

```diff
@@ -59,46 +62,276 @@ compiler.blacklist *gcc*
 post-patch {
+    # Set the default MRUSTC_LIBDIR
     reinplace "s|@MACPORTS_MRUSTC_LIBDIR@|${prefix}/lib/mrustc-${rust_version_major}|" ${worksrcpath}/src/main.cpp
-}
-
-# Set the default CC
-patchfiles-append patch-mrustc-macports-compiler.diff
-post-patch {
+    # Set the default CC
     reinplace "s|@MACPORTS_MRUSTC_CC@|${configure.cc}|" ${worksrcpath}/src/trans/codegen_c.cpp
-}
-
-# Set the default gTargetVersion
-patchfiles-append patch-mrustc-macports-version.diff
-post-patch {
+    # Set the default gTargetVersion
     reinplace "s|@MACPORTS_MRUSTC_TARGET_VER@|${rust_version_major_underscore}|" ${worksrcpath}/src/main.cpp
-}
 
-# Patch the Rust source code with the patch included with mrustc
-post-patch {
+    # Patch the Rust source code with the patch included with mrustc
     system -W ${workpath}/rustc-${rust_version}-src "patch -p0 < ${worksrcpath}/rustc-${rust_version}-src.patch"
     file rename ${workpath}/rustc-${rust_version}-src ${worksrcpath}
+    # prevent it from re-download src
+    touch ${worksrcpath}/rustc-${rust_version}-src/dl-version
 }
 
-build {
-    system -W ${worksrcpath} "/usr/bin/env CXX=${configure.cxx} CC=${configure.cc} CXXFLAGS_EXTRA='${configure.cxxflags}' ${build.cmd} all"
-    system -W ${worksrcpath}/tools/minicargo "/usr/bin/env CXX=${configure.cxx} CC=${configure.cc} ${build.cmd} all"
-    system -W ${worksrcpath} "/usr/bin/env CXX=${configure.cxx} CC=${configure.cc} RUSTC_VERSION=${rust_version} RUSTC_TARGET=${build_arch}-apple-darwin MRUSTC_LIBDIR=output-${rust_version} MRUSTC_TARGET_VER=${rust_version_major} ${build.cmd} -f minicargo.mk LIBS"
+depends_build-append \
+                port:cmake \
+                port:git \
+                port:pkgconfig
+
+depends_lib-append \
+                port:curl \
+                port:libgit2 \
+                port:nghttp2 \
+                port:zlib
+
+# use the system python27 if present
+if {${os.platform} eq "darwin" && ${os.major} >= 11} {
+    set pythonfullpath   /usr/bin/python2.7
+} else {
+    set pythonfullpath   ${prefix}/bin/python2.7
+    depends_build-append port:python27
 }
 
+openssl.branch  1.0
+
+set legacy_features ""
+set legacy_cmake_opts ""
+
+# macOS 10.7 required to use MacPorts clang with emulated TLV
+if {${os.platform} eq "darwin" && ${os.major} < 11} {
+    # __cxa_thread_atexit is part of libcxx, link against it!
+    configure.cxx_stdlib \
+                libc++
+
+    patchfiles-append \
+                0009-macOS-10.6-emulated-TLS-before-10.7.patch \
+                0010-macOS-10.6-fix-build-issues-pre-Lion.patch \
+                0011-macOS-10.6-fix-pthread_cond_destroy.patch \
+                0012-macOS-10.6-support-legacy-crypto.patch
+
+    set legacy_features ",backtrace/std"
+
+    depends_build-append \
+                port:libtool
+
+    set legacy_cmake_opts " CMAKE_LIBTOOL=${prefix}/bin/libtool"
+
+}
+
+# #See: https://github.com/thepowersgang/mrustc/issues/225
+# use_parallel_build no
```

Comment:
Parallel builds are back on?

Reply:
@evanmiller nope, thanks.

Reply:
It may works and depends on number of used CPU. But usually it fails ;)